packages/formstr-app/src/containers/Dashboard/FormCards/FormEventCard.tsx

```diff
@@ -75,32 +76,70 @@ export const FormEventCard: React.FC<FormEventCardProps> = ({
     downloadHTMLToDevice(formFillerUI, name[1]);
   };
 
+  const saveAndOpen = (duplicatedTags: Tag[], newFormId: string) => {
+    const duplicatedForm = {
+      formSpec: duplicatedTags,
+      tempId: newFormId,
+    };
+
+    const existingDrafts = localStorage.getItem("formstr:draftForms");
+    let updatedDrafts = existingDrafts ? JSON.parse(existingDrafts) : [];
+    updatedDrafts = [duplicatedForm, ...updatedDrafts];
+    localStorage.setItem("formstr:draftForms", JSON.stringify(updatedDrafts));
+    window.open(
+      constructDraftUrl(duplicatedForm, window.location.origin),
+      "_blank"
+    );
+  };
+
+  const handleDuplicate = () => {
```

Comment:
it's okay to move this here for now, but ideally i wouldn't want quick actions code bloating the formevents card